src/ui/public/vis/request_handlers/courier.js

```diff
@@ -128,30 +128,32 @@ const CourierRequestHandlerProvider = function (Private, courier, timefilter) {
             request.stats(getRequestInspectorStats(requestSearchSource));
 
             requestSearchSource.onResults().then(resp => {
-              searchSource.lastQuery = queryHash;
-
-              request
-                .stats(getResponseInspectorStats(searchSource, resp))
-                .ok({ json: resp });
-
-              searchSource.rawResponse = resp;
-              return _.cloneDeep(resp);
-            }).then(async resp => {
```

Comment:
@timroes Was there a reason to add another `then` here? I removed it, and everything still seems to work correctly. Can you take a look and let me know if you think this looks OK? I'm removing Angular promises from courier, so I need to wrap Angular state updates in `evalAsync`.